### PR TITLE
enable distributed tracing in client, gateaway, scheduler

### PR DESCRIPTION
**Pull request by akihikokuroda**
### Summary
Fix: #841 

### Details and comments
Enable distributed tracing in client, gateway, scheduler, ray components and payload program.

### CODE REVIEW

File: client/requirements.txt [hunk 1]
```diff
@@ -3,12 +3,13 @@ requests>=2.31.0
 importlib-metadata>=5.2.0
 qiskit-terra==0.24.1
 qiskit-ibm-runtime>=0.11.2
-redis>=4.6.0
+redis==4.6.0
```
Comment (psschwei) [inline client/requirements.txt]: will need to revert this change

File: client/quantum_serverless/quantum_serverless.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# This code is a Qiskit project.
+# This code is a Qiskit project.A
```
Comment (psschwei) [inline client/quantum_serverless/quantum_serverless.py:1]:
```suggestion
# This code is a Qiskit project.
```

File: infrastructure/docker/Dockerfile-gateway [hunk 3]
```diff
@@ -15,6 +15,7 @@ RUN apt-get -y update &&\
 USER 0
 COPY gateway/requirements.txt .
 RUN pip install -r requirements.txt --no-cache-dir
+RUN pip install opentelemetry-distro opentelemetry-exporter-otlp
```
Comment (psschwei) [inline infrastructure/docker/Dockerfile-gateway:18]: are these not picked up for install via the previous step?

**Review by psschwei — COMMENTED** [review]
couple of nits, otherwise LGTM

File: infrastructure/docker/Dockerfile-gateway (hunk 3 above)
Comment (akihikokuroda) [reply infrastructure/docker/Dockerfile-gateway:18]: Thanks! I forgot to remove this line after I updated the requirements.txt

**Review by psschwei — APPROVED** [review]

**Comment by akihikokuroda** [comment]
I believe that this PR can close #841 for the  most part.  Some minor follow on PRs are probably necessary though.

**Review by psschwei — APPROVED** [review]

File: client/quantum_serverless/core/tracing.py [hunk 4]
```diff
@@ -29,7 +29,7 @@
 import os
 from typing import Dict, Optional
 
-from opentelemetry import trace
+from opentelemetry import trace  # pylint: disable=duplicate-code
```
Comment (Tansito) [inline client/quantum_serverless/core/tracing.py:32]: One question here @akihikokuroda , why do we need to add this `disable` option? It seems a false positive but just to be sure.

Comment (akihikokuroda) [reply client/quantum_serverless/core/tracing.py:32]: It complains about 2 files having the similar set of important statements.

**Review by Tansito — APPROVED** [review]
LGTM!

**Comment by Tansito** [comment]
@akihikokuroda let me know if you need help with the conflicts, it should be as easier as copy & paste the content in:
- `charts/quantum-serverless/charts/gateway/templates/deployment.yaml`
- `charts/quantum-serverless/charts/gateway/values.yaml`
- `charts/quantum-serverless/charts/jupyter/templates/deployment.yaml`
- `charts/quantum-serverless/charts/jupyter/values.yaml`
